AzureMonitorAgent/agent.py

```diff
@@ -367,7 +367,7 @@ def install():
 
         if "certificatePath" in protected_settings:
             try:
-                with open(protected_settings.get("certificatePath"), 'r') as f:
+                with open(protected_settings.get("certificatePath"), 'rb') as f:
```

Comment:
This code changes the read path, however the error message provided in the PR points to it attempting to write, which happens here:

https://github.com/Azure/azure-linux-extensions/blob/a81f31479ceaf53742eaeb9fe16af8c9a29b3203/AzureMonitorAgent/agent.py#L452

Reply:
Ah, I see, it reads it and then writes it. Nevermind on this comment.